service/datastore/internal/src/main/java/org/eclipse/kapua/service/datastore/internal/ClientInfoRegistryServiceImpl.java

```diff
@@ -134,8 +133,13 @@ public ClientInfoListResult query(ClientInfoQuery query)
 
         ArgumentValidator.notNull(query, QUERY);
         ArgumentValidator.notNull(query.getScopeId(), QUERY_SCOPE_ID);
-
+        if (query.getLimit() != null && query.getOffset() != null) {
```

Comment:
could you find a way to make this code common to Channel and Messaege instead of replicating it?